### PR TITLE
Updated watchlist

### PR DESCRIPTION
**Pull request by jwalla13**

### CODE REVIEW

File: src/Pages/Dashboard.js [hunk 1]
```diff
@@ -15,37 +16,7 @@ import Button from '@material-ui/core/Button';
...
-      },
-      {
-        ticker: "SHEESH",
-        price: 46.7
-      },
-      {
-        ticker: "OVER",
-        price: .73
-      }
-    ]
```
Comment (jrauvola) [inline src/Pages/Dashboard.js]: Do we want to leave a mock display for when the watchlist is loading?

File: src/Util/apiClient.js [hunk 2]
```diff
@@ -19,6 +19,61 @@ const apiClient = {
...
+    axios.get(apiUrl)
+      .then( res => {
+          if (res){
+            this.getWatchlist(username, setWatchlist)
+          }
+        }
+      )
+      .catch(function (error) {
+          console.log(error);
+      });
```
Comment (jrauvola) [inline src/Util/apiClient.js:76]: LGTM

File: src/Util/apiClient.js [hunk 3]
```diff
@@ -19,6 +19,61 @@ const apiClient = {
...
+
+  getPortfolio: function(username, setPortfolio) {
+    const apiUrl = 'http://localhost:5000/g_prof/' + username;
+    axios.get(apiUrl)
+      .then(res => {
+        const stockListRes = Object.values(res.data);
+        console.log(stockListRes);
+        setPortfolio({list: stockListRes});
+      })
+      .catch(function (error) {
```
Comment (jrauvola) [inline src/Util/apiClient.js:32]: Nice error checking! Make sure to do for other functions as well

File: src/Components/Portfolio.js [hunk 4]
```diff
@@ -84,7 +73,7 @@ function Portfolio(props) {
     apiClient.getStock(ticker, props.setCurrentStock, "detailed")
   }
 
-    if (props.loggedIn) {
+    if (props.user.loggedIn && stockList != null) {
```
Comment (nhans32) [inline src/Components/Portfolio.js:76]: Really like the idea of passing an entire user object!  Good job!

File: src/Pages/Dashboard.js [hunk 5]
```diff
@@ -13,21 +14,35 @@ import BuyPrompt from '../Components/BuyPrompt'
 import Button from '@material-ui/core/Button';
 
 function DashboardPage(props) {
+  const [userInfo,setUserInfo]= useState({
+    username: props.user.username,
+    loggedIn: props.user.loggedIn,
+  })
```
Comment (WesReynolds) [inline src/Pages/Dashboard.js:20]: Not sure that we need to keep track of whether or not the user is currently logged in. If the user is not logged in they cannot press any of the buttons to trigger these methods.

File: src/Util/apiClient.js (hunk 2 above)
Comment (WesReynolds) [inline src/Util/apiClient.js:76]: Nice clean code. LGTM

File: src/App.js [hunk 6]
```diff
@@ -66,7 +66,7 @@ class App extends Component {
               exact
               path='/dashboard' 
               render={props => (
-                <Dashboard {...props} loggedIn={this.state.loggedIn} loggedUsername={this.state.username} loggedId={this.state.id}/>
+                <Dashboard {...props} user={this.state} loggedIn={this.state.loggedIn} loggedUsername={this.state.username} loggedId={this.state.id}/>
```
Comment (nhans32) [inline src/App.js:69]: do we need to retain loggedUsername/loggedIn as a passed prop if we are passing a user object already?

File: src/Util/apiClient.js (hunk 2 above)
Comment (nhans32) [reply src/Util/apiClient.js:76]: LGTM

**Review by nhans32 — APPROVED** [review]
Nice!

File: src/Components/Portfolio.js (hunk 4 above)
Comment (jwalla13) [reply src/Components/Portfolio.js:76]: Thanks!

File: src/Pages/Dashboard.js (hunk 5 above)
Comment (jwalla13) [reply src/Pages/Dashboard.js:20]: Good point, also the username will be null if we ever do need to determine if they're logged in or not

File: src/App.js (hunk 6 above)
Comment (jwalla13) [reply src/App.js:69]: Oops I meant to remove all of those when I updated to passing a user object, good catch!

File: src/Pages/Dashboard.js (hunk 1 above)
Comment (jwalla13) [reply src/Pages/Dashboard.js]: The stocks wouldn't be accurate for the user which could make the experience a little confusing

File: src/Util/apiClient.js (hunk 3 above)
Comment (jwalla13) [reply src/Util/apiClient.js:32]: Good call out, should definitely be error checking for each api call.

**Pull request merged**